templates/cloud_init/cloud_init.client.yml

```
@@ -0,0 +1,367 @@
+      [ $? -ne 0 ] && error "sshkey."
+
+      cd ../vdbench
+
+      terraform init
+      TF_LOG=INFO terraform destroy -auto-approve -var="location=${REGION}" \
+        -var="vmss_resource_group_name=${RESOURCE_GROUP_NAME}-vmss" \
+        -var="controller_address=${CONTROLLER_ADDRESS}" \
+        -var="controller_username=${USERNAME}" \
+        -var="ssh_key_data=${sshkey}" \
+        -var="vserver_ip_addresses=${VSERVER_IP_ADDRESSES}" \
+        -var="nfs_export_path=${NFS_EXPORT_PATH}" \
+        -var="vdbench_url=<vdbench_download_url>" \
+        -var="vnet_resource_group=${RESOURCE_GROUP_NAME}-network"
+
+  - path: /opt/vdbench_run.sh
+    permissions: '0755'
+    content: |
+      #!/bin/bash
+
+      function error() {
+        >&2 echo "echo >< ERROR: $* See log for details."
+        exit 1
+      }
+
+      export AZURE_SUBSCRIPTION='<sub_id>'
+      export ARM_SUBSCRIPTION_ID='<sub_id>'
+      export ARM_CLIENT_ID='<arm_client_id>'
+      export ARM_CLIENT_SECRET='<arm_client_scrt>'
+      export ARM_TENANT_ID='<arm_tenant_id>'
+      export REGION='<region>'
+      export RESOURCE_GROUP_NAME='<resource_group_name>'
+      export RUN_TERRAFORM_FILE='<run_terraform_file>'
+
+      echo "az login"
+      az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID
+      [ $? -ne 0 ] && error "az login."
+
+      echo "az acct set"
+      az account set -s $AZURE_SUBSCRIPTION
+      [ $? -ne 0 ] && error "az acct set."
+
+      echo "to file path"
+      cd /tf/src/terraform/examples/vfxt/${RUN_TERRAFORM_FILE}
+      [ $? -ne 0 ] && error "filepath."
+
+      echo "terraform outputs"
+      CONTROLLER_ADDRESS=$(terraform output -json | jq -r ".controller_address.value")
+      USERNAME=$(terraform output -json | jq -r ".controller_username.value")
+      VSERVER_IP_ADDRESSES=$(terraform output -json | jq -c ".mount_addresses.value")
+      NFS_EXPORT_PATH=$(terraform output -json | jq -r ".mount_namespace_path.value")
+      [ $? -ne 0 ] && error "terraform outputs."
+
+      echo "sskey"
+      sshkey="<ssh_key> azureuser@linuxvm"
+      [ $? -ne 0 ] && error "sshkey."
+
+      cd ../vdbench
+
+      terraform init
+      TF_LOG=INFO terraform apply -auto-approve -var="location=${REGION}" \
+        -var="vmss_resource_group_name=${RESOURCE_GROUP_NAME}-vmss" \
+        -var="controller_address=${CONTROLLER_ADDRESS}" \
+        -var="controller_username=${USERNAME}" \
+        -var="ssh_key_data=${sshkey}" \
+        -var="vserver_ip_addresses=${VSERVER_IP_ADDRESSES}" \
+        -var="nfs_export_path=${NFS_EXPORT_PATH}" \
+        -var="vdbench_url=<vdbench_download_url>" \
+        -var="vnet_resource_group=${RESOURCE_GROUP_NAME}-network"
+
+      echo "vdbench terraform outputs"
+      IP_ADDRESS_COMMAND=$(terraform output -json | jq -r ".vmss_addresses_command.value")
+      IP_ADDRESS=$(${IP_ADDRESS_COMMAND} | jq -r '.[0]')
+      [ $? -ne 0 ] && error "vdbench terraform outputs."
```

If you want to check multiple commands with one error check, you can chain the commands with `&&`.
```suggestion
      IP_ADDRESS_COMMAND=$(terraform output -json | jq -r ".vmss_addresses_command.value") &&
      IP_ADDRESS=$(${IP_ADDRESS_COMMAND} | jq -r '.[0]')
      [ $? -ne 0 ] && error "vdbench terraform outputs."
```

---

Updated this!